datafusion/core/src/physical_optimizer/combine_partial_final_agg.rs

```diff
@@ -86,7 +86,7 @@ impl PhysicalOptimizerRule for CombinePartialFinalAggregate {
                                             } else {
                                                 AggregateMode::SinglePartitioned
                                             };
-                                        AggregateExec::try_new(
+                                        let combined_agg = AggregateExec::try_new(
```

Comment:
Another way to express the same logic with less indenting is:

```rust
                                        AggregateExec::try_new(
                                            mode,
                                            input_agg_exec.group_by().clone(),
                                            input_agg_exec.aggr_expr().to_vec(),
                                            input_agg_exec.filter_expr().to_vec(),
                                            input_agg_exec.order_by_expr().to_vec(),
                                            input_agg_exec.input().clone(),
                                            input_agg_exec.input_schema().clone(),
                                        )
                                        .map(|combined_agg| {
                                            combined_agg.with_limit(agg_exec.limit())
                                        })
                                        .ok()
                                        .map(Arc::new)
```

Reply:
> Another way to express the same logic with less indenting is:
> 
> ```rust
>                                         AggregateExec::try_new(
>                                             mode,
>                                             input_agg_exec.group_by().clone(),
>                                             input_agg_exec.aggr_expr().to_vec(),
>                                             input_agg_exec.filter_expr().to_vec(),
>                                             input_agg_exec.order_by_expr().to_vec(),
>                                             input_agg_exec.input().clone(),
>                                             input_agg_exec.input_schema().clone(),
>                                         )
>                                         .map(|combined_agg| {
>                                             combined_agg.with_limit(agg_exec.limit())
>                                         })
>                                         .ok()
>                                         .map(Arc::new)
> ```

Applied this change from your example PR, thanks!